todo_app/app.py

```diff
@@ -1,36 +1,107 @@
+import os
 from flask import Flask, render_template, request, redirect
-from datetime import datetime
-from todo_app.data.session_items import check_default_lists, create_card, get_lists, get_cards, move_card, remove_card, ViewModel
+import requests
+from flask_login import LoginManager, login_required, login_user, current_user
+from oauthlib.oauth2 import WebApplicationClient
+from loggly.handlers import HTTPSHandler
+from logging import Formatter
+from todo_app.data.session_items import (
+    check_default_lists, create_card, get_lists, get_cards,move_card,
+    remove_card, ViewModel
+)
+from todo_app.user import TestUser,User,Anonymous
+
+login_manager = LoginManager()
+
+@login_manager.unauthorized_handler
+def unauthenticated():
+    github_client =  WebApplicationClient(os.environ.get('OAUTH_CLIENT_ID'))
+    github_redirect = github_client.prepare_request_uri("https://github.com/login/oauth/authorize")
+
+    return redirect(github_redirect)
+
+@login_manager.user_loader
+def load_user(user_id):
+    return User(user_id)
 
 def create_app():
     app = Flask(__name__)
+    app.secret_key = os.getenv('OAUTH_CLIENT_SECRET')
+    login_manager.init_app(app)
+    if os.getenv('FLASK_ENV') == 'e2e_Test':
+        login_manager.anonymous_user = TestUser
+    else:
+        login_manager.anonymous_user = Anonymous
+
+    app.config['LOG_LEVEL'] = os.getenv('LOG_LEVEL')
```

Comment:
You could set a default value here:
```suggestion
    app.config['LOG_LEVEL'] = os.getenv('LOG_LEVEL', 'INFO')
```